CODINGSTYLE.md

```diff
@@ -22,6 +22,68 @@ In these cases, it can be overridden by surrounding it with `clang-format-off` a
 * Getter functions use the name without "m": `data()`.
 * Setter functions use the "set" prefix: `set_data(value)`.
 
+### Coding patterns
+
+This section describes some of the coding patterns we prefer to see in our code.
+Using the same kind of coding patterns consistently makes it easier for others to understand the code at first glance.
+Conversely, code that breaks the usual patterns takes more time to understand and may put the reader on the wrong foot.
+
+Exceptions are not used at all in prplMesh code.
+If a standard library function is called that may throw an exception, make sure to do the necessary checks before, to make sure an exception cannot be thrown.
+
+#### Loops
+
+Avoid endless loops - some static analysis tools give an error or have a hard time analysing it.
+They are usually able to analyse some forms of endless loops, but it's better to be consistent and never have them.
+Example:
+
+```cpp
+    static const size_t max_tlv = 256;
+    size_t tlv_idx;
+    for (tlv_idx = 0; tlv_idx < max_tlv; tlv_idx++) {
```

Comment:
I would give a bad example like `while (true)`

Reply:
In general I prefer to avoid bad examples unless you can make it visually *very* clear that it's bad.

I will add "like `while (true)`" in the text, that would work.